Issue 5598 - (2nd) In 2.x, SRCH throughput drops by 10% because of ha…

tbordaz opened the pull request:
…ndling of referral

progier389 reviewed (approved):
LGTM

mreynolds389 reviewed (approved):
LGTM - basic and filter tests pass

tbordaz committed:
Issue 5598 - (2nd) In 2.x, SRCH throughput drops by 10% because of handling of referral

Bug description:
        The first fix 5598 introduce/reveal a leak.
	My initial understanding of SLAPI_SEARCH_FILTER and
	SLAPI_SEARCH_FILTER_INTENDED was wrong.
	Without referral, they are identical (refering to the same
        filter). In case of referral, SLAPI_SEARCH_FILTER is
        a craft one that *includes* the original
	(SLAPI_SEARCH_FILTER_INTENDED).

Fix description:
        If there is no referral, SLAPI_SEARCH_FILTER_INTENDED
	and SLAPI_SEARCH_FILTER are just identical

    relates:  #5598

    Reviewed by: Mark Reynolds, Pierre Rogier(thanks)

mreynolds389 reviewed (approved):
Much better!  Ack
